concrete/controllers/single_page/dashboard/system/express/entities/forms.php

```diff
@@ -316,15 +320,31 @@ public function edit($id = null)
         }
     }
 
+    protected function canDeleteForm(Entity $entity, Form $form)
+    {
+        $canDeleteForm = true;
+        if (
+            ($entity->getDefaultEditForm() && $entity->getDefaultEditForm()->getID() == $form->getID()) ||
+            ($entity->getDefaultViewForm() && $entity->getDefaultViewForm()->getID() == $form->getID())
```

Comment:
I'm 99% sure that Doctrine reuse entity instances, so it could be enough to have
```
if (
    $entity->getDefaultEditForm() === $form ||
    $entity->getDefaultViewForm() === $form)
```